### PR TITLE
Provide alternative implementation for get_modules_defined_attrs

### PR DESCRIPTION
**Pull request by adrianparvino**

### CODE REVIEW

File: nixui/nix_eval.py [hunk 1]
```diff
@@ -24,22 +24,41 @@ def nix_instantiate_eval(expr, strict=False):
...
-                )
-                if not res:
-                    attributes[tuple(full_attr_loc)] = sub_attr_datapoint
-                else:
-                    attributes.update(res)
-    return attributes
 
+    return {
+        tuple(v['name']): {"position": v['position']}
+        for v in nix_instantiate_eval("""
```
Comment (lapp0) [inline nixui/nix_eval.py:30]: nit: we shouldn't have huge strings within generators.

```
result = nix_instantiate_eval("""
...
...
""")
return {tuple(v['name']): {"position": v['position']} for v in result}
```

for

File: nixui/nix_eval.py [hunk 2]
```diff
@@ -24,22 +24,41 @@ def nix_instantiate_eval(expr, strict=False):
...
+let
+  config = import <nixos-config> {config = {}; pkgs = import <nixpkgs> {}; lib = import <nixpkgs/lib>;};
```
Comment (lapp0) [inline nixui/nix_eval.py:32]: I'm not sure we should hardcode `nixos-config` here. Eventually we'll want to 
- 1) get attribute positions for each module within `imports` (excluding those within `nixpkgs`, e.g. `"${nixpkgs}/nixos/modules/installer/cd-dvd/iso-image.nix"`)
- 2) get attribute positions for non-configuration files where users want to modify the configurations of other build types much further down the line (e.g. https://github.com/lapp0/nix-gui/issues/23)

**Comment by lapp0** [comment]
Part of https://github.com/lapp0/nix-gui/issues/9

**Comment by adrianparvino** [comment]
Small followup: I think we should extract all inline Nix expressions to its own separate file.

File: nixui/nix_eval.py (hunk 2 above)
Comment (adrianparvino) [reply nixui/nix_eval.py]: See 9bd09a41c8f3927c6e8c98e30a7ec7cf07c789a3

File: nixui/nix_eval.py (hunk 1 above)
Comment (adrianparvino) [reply nixui/nix_eval.py]: See a3bfbd0889c92dd01b91eaf3c160abe7a04d3e33

**Comment by lapp0** [comment]
Yes, I was just thinking that. Additionally, considering the fact that nix uses so many curly-braces we should use a templating engine dissimilar to f-strings so 
`{config = {}; pkgs = import <nixpkgs> {}; lib = import <nixpkgs/lib>;}`
doesn't need to be 
`{{config = {{}}; pkgs = import <nixpkgs> {{}}; lib = import <nixpkgs/lib>;}}`

**Pull request merged**